lib/active_record_shards/default_slave_patches.rb

```diff
@@ -25,15 +25,16 @@ def #{method}_with_default_slave#{punctuation}(*args, &block)
       RUBY
     end
 
-    def columns_with_default_slave(*args, &block)
-      read_columns_from =
-        if on_slave_by_default? && !Thread.current[:_active_record_shards_slave_off]
```

Comment:
Don't we need to check the current Thread anymore?

Reply:
nope ... forcing it even if inside a thread